backend/clickhouse/migrations/000012_create_traces_table.up.sql

```diff
@@ -0,0 +1,40 @@
+
+CREATE TABLE IF NOT EXISTS traces
+(
+    ProjectId UInt32,
+    SecureSessionId String,
+    Timestamp DateTime,
+    TraceId String,
+    SpanId String,
+    ParentSpanId String,
+    TraceState String,
+    SpanName LowCardinality(String),
+    SpanKind LowCardinality(String),
+    Duration Int64,
+    ServiceName LowCardinality(String),
+    ResourceAttributes Map(LowCardinality(String), String),
+    SpanAttributes Map(LowCardinality(String), String),
+    StatusCode LowCardinality(String),
+    StatusMessage String,
+    Events Nested (
+        Timestamp DateTime,
+        Name LowCardinality(String),
+        Attributes Map(LowCardinality(String), String)
+    ),
+    Links Nested (
+        TraceId String,
+        SpanId String,
+        TraceState String,
+        Attributes Map(LowCardinality(String), String)
+    ),
+    INDEX idx_trace_id TraceId TYPE bloom_filter(0.001) GRANULARITY 1,
+    INDEX idx_res_attr_key mapKeys(ResourceAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
+    INDEX idx_res_attr_value mapValues(ResourceAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
+    INDEX idx_span_attr_key mapKeys(SpanAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
+    INDEX idx_span_attr_value mapValues(SpanAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
+    INDEX idx_duration Duration TYPE minmax GRANULARITY 1
+)
+    ENGINE = MergeTree
+        PARTITION BY toDate(Timestamp)
+        ORDER BY (ServiceName, SpanName, toUnixTimestamp(Timestamp), TraceId)
+        SETTINGS index_granularity = 8192, ttl_only_drop_parts = 1;
```

Comment:
Most of this migration came from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/f5275de9a1355c13d85161db6b2eae9d239f1ead/exporter/clickhouseexporter/exporter_traces.go#L177-L212.

Reply:
Let's more closely replicate logs since we know that works correctly for our setup. Check out `backend/clickhouse/migrations/000006_create_logs_new.up.sql`

Namely, the following sticks out to me:

* the OTEL schema doesn't handle multi-tenancy, so `ProjectId` should in the `ORDER BY` since it'll always be part of our querying (see [best practices](https://clickhouse.com/docs/en/optimize/sparse-primary-indexes))
* Add a UUID column so we can handle cursor pagination (see [blog](https://www.highlight.io/blog/how-we-built-logging-with-clickhouse)).
* Merge `ResourceAttributes` and `SpanAttributes` into a single column (similar to `LogAttributes) -- would recommend calling the merged column `TraceAttributes` for consistency.
* Add a column for `ServiceVersion` (see #6117)

Reply:
cc @mayberryzane for thoughts on the `partition by` and `order by`, as he's been doing some digging into clickhouse query perf.

Reply:
I think a custom `partition by` is only helpful if we also have a TTL on this table - are we going to add a 30 day TTL like we do for logs? May also be good to test performance - e.g. without partition, deleting will be slower but querying could be faster?
For the `order by` - I think it may make sense to have it be similar to logs (`ProjectId, Timestamp, UUID`) if the default view is going to be showing all spans in the project. Unless we're always filtering by `ServiceName` and `SpanName` but these seem like secondary filters we could add on to the default

Reply:
I updated the `order by` to only include `ProjectId`, `Timestamp`, and `UUID`. I also set a 30 day TTL, similar to the logs table. Let me know if you see anything else you think we should tweak here before merging.

Reply:
could we remove `PARTITION BY` altogether and see how that performs?